### ISSUE TITLE
[FEATURE REQUEST] Community support

### ISSUE THREAD

**Issue by unilogica** [WhiskeySockets/Baileys#255]
I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?

https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts

**Comment by Auties00** [comment]
> I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> 
> https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts

I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done

**Comment by unilogica** [comment]
Wow! I started reading the documentation of your Cobalt repo, what a fantastic job 👏

**Comment by Auties00** [comment]
Uhm I'm having some problems with the community create node. It looks right, but it's throwing:
Received node Node[description=iq, attributes={from=g.us, id=d329cc8a1b, type=error}, content=[Node[description=error, attributes={code=400, text=bad-request}]]]
I'll try to fix it asap so I can open the PR

**Comment by Auties00** [comment]
> I saw some of your repos and noticed the WhatsappWebRequestAnalyzer. Java and Kotlin aren't even close to my most familiar stacks, but from what I understand this project uses Chrome 90 injected with a layer that takes every request made by WhatsApp.
> 
> Is that how you were able to reverse engineer this community feature originally for Cobalt?

Yeah, actually I use chrome 114, but you should be able to use any Chrome ≥ 111
To give you a gist, the analyzer:
1. Starts an instance of Chrome
2. Goes to Whatsapp web
3. Instantly finds the line where the crypto keys are loaded and places a breakpoint there
4. As the page gets loaded immediately after, the breakpoint is triggered and the keys are extracted
5. I then listen for all web socket events, incoming and outgoing, to decode the Nodes

You also need to understand the original Whatsapp web source code most of the times, but having the raw node is definitely a good start.

To see the mobile's app nodes download the Whatsapp beta and run the following adb command:
adb shell logcat | grep WhatsApp: 
You might need an APK from may 2023 because afaict new versions don't print them out.

**Comment by guiguicdd** [comment]
Guys. I used this code to make the IQ request. Bug getting 400. Any Idea?

```

communityCreate: async (subject: string) => {
  const result = await groupQuery("@g.us", "set", [
    {
      tag: "create",
      attrs: {
        subject,
      },
      content: [
        {
          tag: "description",
          attrs: {
            id: generateMessageTag(),
          },
          content: [
            {
              tag: "body",
              attrs: {},
              content: Buffer.from("The community description", "utf-8"),
            },
          ],
        },
        {
          tag: "parent",
          attrs: {
            default_membership_approval_mode: "request_required",
          },
        },
      ],
    },
  ]);

  console.log(result);

  // return extractGroupMetadata(result)
};

```

**Comment by tizum** [comment]
any news on this subject? it would be great to be able to count communities so you can message them mainly in the general notices group.

**Comment by luizsvl** [comment]
Up

**Comment by Auties00** [comment]
If someone still needs it I could port it from my library as I found the fix

**Comment by iannortnau** [comment]
Any updates on the matter, I would really like to use the community functions, I saw that @Auties00 had found a solution, but has it already been incorporated into Baileys?

**Comment by Auties00** [comment]
> Any updates on the matter, I would really like to use the community functions, I saw that @Auties00 had found a solution, but has it already been incorporated into Baileys?

I should add support for both communities and channels in the near future. I'd like to do it this week, but I have to study as my mid terms are coming up for uni. I'll do it asap tho

**Comment by iannortnau** [comment]
> > Any updates on the matter, I would really like to use the community functions, I saw that @Auties00 had found a solution, but has it already been incorporated into Baileys?
> 
> I should add support for both communities and channels in the near future. I'd like to do it this week, but I have to study as my mid terms are coming up for uni. I'll do it asap tho

Thank you for the response and the attention, I will be waiting and good luck with your studies.

**Comment by Auties00** [comment]
> > > Any updates on the matter, I would really like to use the community functions, I saw that @Auties00 had found a solution, but has it already been incorporated into Baileys?
> > 
> > I should add support for both communities and channels in the near future. I'd like to do it this week, but I have to study as my mid terms are coming up for uni. I'll do it asap tho
> 
> Thank you for the response and the attention, I will be waiting and good luck with your studies.

Thanks :)

**Comment by luanhduarte** [comment]
I'm looking forward to receive this new feature in baileys!!!

**Comment by luangduartedev** [comment]
Are there some news?

**Comment by luispimenta** [comment]
Are there some news to receive this new feature in baileys?

**Comment by SatzzDev** [comment]
any update?

**Comment by joaosouz4dev** [comment]
@SatzzDev @luispimenta @luangduartedev @luanhduarte @Auties00 @iannortnau @luizsvl @tizum @guiguicdd 

I created a support for communities, can you test it?

**Comment by AstroX11** [comment]
> > I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> > https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts
> 
> I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done

Tomorrow never came

**Comment by joaosouz4dev** [comment]
> > > I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> > > https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts
> > 
> > 
> > I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done
> 
> Tomorrow never came

can u start here https://github.com/WhiskeySockets/Baileys/pull/1157

**Comment by AstroX11** [comment]
> > > > I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> > > > https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts
> > > 
> > > 
> > > I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done
> > 
> > 
> > Tomorrow never came
> 
> can u start here [#1157](https://github.com/WhiskeySockets/Baileys/pull/1157)

okay

**Comment by AstroX11** [comment]
> > > > I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> > > > https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts
> > > 
> > > 
> > > I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done
> > 
> > 
> > Tomorrow never came
> 
> you don't have the right to say that you leecher

leecher? bro must be retarded, it seems you're very verbose and mentally incapable

**Comment by joaosouz4dev** [comment]
> > > > > I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> > > > > https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts
> > > > 
> > > > 
> > > > I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done
> > > 
> > > 
> > > Tomorrow never came
> > 
> > 
> > you don't have the right to say that you leecher
> 
> leecher? bro must be retarded, it seems you're very verbose and mentally incapable

Calm down guys, regarding this topic of using wppconnect codes for this, I believe that is not the case, the libraries are very different, I personally would spend a few hours trying to explain how to do this reverse engineering that Baileys does and maybe I wouldn't be able to explain what I understood.
note: I am a founder of wppconnect

**Comment by AstroX11** [comment]
> > > > > > I would suggest community support. I've seen WPPConnect implement this and would like to take the opportunity and ask if anyone knows how I can port this to the Baileys. Exists some doc to explain how can I inspect WhatsApp to create this feature?
> > > > > > https://github.com/wppconnect-team/wppconnect/blob/master/src/api/layers/community.layer.ts
> > > > > 
> > > > > 
> > > > > I can probably port it from my library cobalt tomorrow. I'll open a PR once I'm done
> > > > 
> > > > 
> > > > Tomorrow never came
> > > 
> > > 
> > > you don't have the right to say that you leecher
> > 
> > 
> > leecher? bro must be retarded, it seems you're very verbose and mentally incapable
> 
> Calm down guys, regarding this topic of using wppconnect codes for this, I believe that is not the case, the libraries are very different, I personally would spend a few hours trying to explain how to do this reverse engineering that Baileys does and maybe I wouldn't be able to explain what I understood. note: I am a founder of wppconnect

I understand you're work, and there's no need to use wpp codes your PR is enough, and secondly bro's calling me a code thief and he doesn't who I am crazy troll
